aten/src/ATen/native/cuda/Unique.cu

```diff
@@ -15,148 +16,211 @@ namespace native{
 
 namespace {
 
+
+template <
+  typename policy_t, typename scalar_t,
+  typename equal_t, typename not_equal_t
+>
+std::tuple<Tensor, Tensor, int64_t> compute_unique(
+  const policy_t &policy,
+  scalar_t *data,
+  int64_t num_inp,
+  int64_t *sorted_indices_ptr,
+  const bool return_inverse,
+  const bool return_counts,
+  TensorOptions options,
+  equal_t equal,
+  not_equal_t not_equal
+) {
+
+  // inverse indices
+  Tensor inverse_indices;
+  if (!return_inverse) {
+    inverse_indices = at::empty({0}, options);
+  } else {
+    Tensor inv_loc = at::empty({num_inp}, options);
+    inverse_indices = at::empty({num_inp}, options);
+    int64_t* inv_loc_ptr = inv_loc.data<int64_t>();
+    int64_t* inverse_indices_ptr = inverse_indices.data<int64_t>();
+    thrust::adjacent_difference(policy, data, data + num_inp, inv_loc_ptr, not_equal);
+    inv_loc[0] = 0;
+    thrust::inclusive_scan(policy, inv_loc_ptr, inv_loc_ptr + num_inp, inv_loc_ptr);
+    thrust::scatter(policy, inv_loc_ptr, inv_loc_ptr + num_inp, sorted_indices_ptr, inverse_indices_ptr);
```

Comment:
I'd sleep better if there was an assert for sorted_indices_ptr being non-null here - it currently works the way your code is structured, but if compute_unique is called in a different way it could break. Better yet, send sorted_indices tensor here, and check if it is .defined()

Reply:
fixed